src/core/credentials/transport/google_default/google_default_credentials.cc

```diff
@@ -445,26 +463,44 @@ grpc_channel_credentials* grpc_google_default_credentials_create(
       << "grpc_google_default_credentials_create(" << call_creds_for_tls << ")";
 
   if (call_creds == nullptr) {
-    call_creds = make_default_call_creds(&error);
+    auto create_default_creds_status = CreateGoogleDefaultCallCredentials();
+
+    if (!create_default_creds_status.ok()) {
+      LOG(ERROR) << "Could not create google default credentials: "
+                 << grpc_core::StatusToString(
+                        create_default_creds_status.status());
+    }
+    call_creds = create_default_creds_status.value().first;
```

Comment:
This needs an `else`, because if `create_default_creds_status` is not OK, then it's not safe to access the value.

Also, you should not use `create_default_creds_status.value()` after you've already checked `ok()`, as per [abseil.io/tips/181](https://abseil.io/tips/181).  Instead, please use `*create_default_creds_status` or `create_default_creds_status->`.